lib/transloadit.rb

```diff
@@ -130,6 +135,58 @@ def to_json
     MultiJson.dump(to_hash)
   end
 
+  # @param workspace [String] Workspace slug
+  # @param template [String] Template slug or template ID
+  # @param input [String] Input value that is provided as `${fields.input}` in the template
+  # @param url_params [Hash] Additional parameters for the URL query string (optional)
+  # @param expire_at_ms [Integer] Expiration time as Unix timestamp in milliseconds (optional)
+  # @return [String] Signed Smart CDN URL
+  def signed_smart_cdn_url(
+    workspace:,
+    template:,
+    input:,
+    expire_at_ms: nil,
+    url_params: {}
+  )
+    raise ArgumentError, "workspace is required" if workspace.nil?
+    raise ArgumentError, "template is required" if template.nil?
+    raise ArgumentError, "input is required" if input.nil?
+
+    workspace_slug = CGI.escape(workspace)
+    template_slug = CGI.escape(template)
+    input_field = CGI.escape(input)
+
+    expire_at = expire_at_ms || (Time.now.to_i * 1000 + 60 * 60 * 1000) # 1 hour default
+
+    query_params = {}
+    url_params.each do |key, value|
+      next if value.nil?
+      if value.is_a?(Array)
+        value.each do |val|
+          next if val.nil?
+          (query_params[key.to_s] ||= []) << val.to_s
+        end
+      else
+        query_params[key.to_s] = [value.to_s]
+      end
```

Comment:
You can use our little `Array` friend here, making the logic the same for both branches.

```suggestion
  Array(value).each do |val|
    next if val.nil?
    (query_params[key.to_s] ||= []) << val.to_s
  end
```

Reply:
Thanks!